google/auth/impersonated_credentials.py

```diff
@@ -288,7 +288,12 @@ def sign_bytes(self, message):
             url=iam_sign_endpoint, headers=headers, json=body
         )
 
-        return base64.b64decode(response.json()["signedBlob"])
+        response_data = response.json()
+
+        if "error" in response:
+          raise Exception(response_data["error"])
```

Comment:
The exception type should be based on the status code from the response (which we ought to be checking before extracting JSON, anyway).

We will also need one or more new tests for the error case.